### ISSUE TITLE
SDDM Greeter crashes in some combination used with kwin

### ISSUE THREAD

**Issue by Svenum** [NixOS/nixpkgs#311450]
### Describe the bug
If kwin is the compositor of sddm you only get a black screen after boot. It worked fine before I updated my nixos flake.

### Steps To Reproduce
1. Enable sddm like this:
    ```nix
    service.displayManager.sddm = {
      enable = true;
      wayland = {
        enable = true;
        compositor = "kwin";
      };
      autoNumlock = true;
    };
    ``` 
2. Build config `nixos-rebuild boot`
3. Reboot system

### Expected behavior
The Login screen should appear and `CTRL + F1` - `CTRL + F6` should give an terminal output.

### Additional context
output of `journalctl -b -1 -u display-manager.service`:
```console
Mai 13 21:06:33 Yon systemd[1]: Started Display Manager.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Mai 13 21:06:33 Yon sddm[1545]: Initializing...
Mai 13 21:06:33 Yon sddm[1545]: Starting...
Mai 13 21:06:33 Yon sddm[1545]: Logind interface found
Mai 13 21:06:33 Yon sddm[1545]: Adding new display...
Mai 13 21:06:33 Yon sddm[1545]: Loaded empty theme configuration
Mai 13 21:06:33 Yon sddm[1545]: Using VT 7
Mai 13 21:06:33 Yon sddm[1545]: Display server started.
Mai 13 21:06:33 Yon sddm[1545]: Socket server starting...
Mai 13 21:06:33 Yon sddm[1545]: Socket server started.
Mai 13 21:06:33 Yon sddm[1545]: Loading theme configuration from "/run/current-system/sw/share/sddm/themes/breeze/theme.conf"
Mai 13 21:06:33 Yon sddm[1545]: Greeter starting...
Mai 13 21:06:33 Yon sddm-helper[1550]: Detected locale "C" with character encoding "ANSI_X3.4-1968", which is not UTF-8.
                                       Qt depends on a UTF-8 locale, and has switched to "C.UTF-8" instead.
                                       If this causes problems, reconfigure your locale. See the locale(1) manual
                                       for more information.
Mai 13 21:06:33 Yon sddm-helper[1550]: [PAM] Starting...
Mai 13 21:06:33 Yon sddm-helper[1550]: [PAM] Authenticating...
Mai 13 21:06:33 Yon sddm-helper[1550]: [PAM] returning.
Mai 13 21:06:33 Yon sddm[1545]: Greeter session started successfully
Mai 13 21:06:35 Yon sddm[1545]: Message received from greeter: Connect
Mai 13 21:06:56 Yon sddm[1545]: Error from greeter session: "Process crashed"
Mai 13 21:06:56 Yon sddm[1545]: Auth: sddm-helper (--socket /tmp/sddm-auth-28334843-3453-40d7-b871-f96a6180049c --id 2 --start /nix/store/4vr0hbjmbnmr8k5gv4mdwfvc36csr6pr-sddm-wrapped/bin/sddm-greeter-qt6 --socket /tmp/sddm--UJfsPi --theme /run/current-system/sw/share/sddm/themes/breeze --user sddm --display-server /nix/store/rizf7g8a5w26lxamz1f5d41knd01pivk-kwin-6.0.4.1/bin/kwin_wayland --no-global-shortcuts --no-kactivities --no-lockscreen --locale1 --greeter) crashed (exit code 1)
Mai 13 21:06:56 Yon sddm[1545]: Error from greeter session: "Process crashed"
Mai 13 21:06:56 Yon sddm[1545]: Auth: sddm-helper exited with 1
Mai 13 21:06:56 Yon sddm[1545]: Greeter stopped. SDDM::Auth::HELPER_AUTH_ERROR
Mai 13 21:06:56 Yon systemd[1]: Stopping Display Manager...
Mai 13 21:06:56 Yon sddm[1545]: Signal received: SIGTERM
Mai 13 21:06:56 Yon sddm[1545]: Socket server stopping...
Mai 13 21:06:56 Yon sddm[1545]: Socket server stopped.
Mai 13 21:06:56 Yon systemd[1]: display-manager.service: Deactivated successfully.
Mai 13 21:06:56 Yon systemd[1]: Stopped Display Manager.
```

### Notify maintainers

@SuperSandro2000 
@abbradar
@ttuegel
@k900
@

### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.8.9, NixOS, 24.05 (Uakari), 24.05.20240512.2057814`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.23.0pre20240502_00ca2b05`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/vsbi6gfwf2bdamw0d0d9xndb1rrxfnbk-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by SuperSandro2000** [comment]
> Detected locale "C" with character encoding "ANSI_X3.4-1968", 

Do you have some none default locale configured?

Do you have some sddm theme?

**Comment by Svenum** [comment]
Yes, sddm with plasma and without theme. As locale (i18n.defaultLocale) I use en_US.UTF-8 and for the other LC_... envs I use de_DE.UTF-8

**Comment by tjkirch** [comment]
I have two machines running NixOS with almost identical configurations, but one has an nVidia graphics card (using proprietary driver) and one has an AMD graphics card (using open source driver).  Only the nVidia machine has this problem.  I'm not sure if that's related or if it's chance.

> > Detected locale "C" with character encoding "ANSI_X3.4-1968",
> 
> Do you have some none default locale configured?

All locale settings are set to `en_US.UTF-8` for me, but I saw this warning as well, on both machines.

> Do you have some sddm theme?
> Are you using sddm with plasma?

The only SDDM settings are to use the KWin compositor and Wayland, with Plasma 6.

**Comment by Svenum** [comment]
Additional Info: My Machine has an AMD GPU

**Comment by SuperSandro2000** [comment]
No idea, works on Intel iGPU.

**Comment by Svenum** [comment]
Could you tell me how to find logs of sddm-helper? Sddm-helper is not an service as I know.

**Comment by SuperSandro2000** [comment]
I couldn't find it either

**Comment by Svenum** [comment]
I updated my nix flakes again `nix flake update` and now its working again....

**Comment by tjkirch** [comment]
> I updated my nix flakes again `nix flake update` and now its working again....<
> 
> So feel free to close if others not effected anymore too.

Sadly, it's no better for me after today's update.

![sddm-crash](https://github.com/NixOS/nixpkgs/assets/13994/6bc5f3cd-cc05-4889-8c02-d1fc463e290c)

I'd be happy to test any debugging theories; I'm new to Nix and KDE.

**Comment by Steinhagen** [comment]
Having the same issue with the proprietary NVidia driver (GTX 980). The timing this issue happened matched this ticket:
https://gist.github.com/Steinhagen/9d4a306acb0d041f6b51f83c1b8f409a

It seems that this issue happens if you either use SDDM with Wayland enabled, or if you start KWin using Wayland.
If you use X11, both SDDM and KWin will work as expected.

**Comment by Steinhagen** [comment]
This issue seems to happen with the latest Linux kernel. Downgrading from **linuxPackages_latest** to **linuxPackages** evades the broken Wayland issue for now.

**Comment by opl-** [comment]
As mentioned [here](https://github.com/NixOS/nixpkgs/issues/323396#issuecomment-2198270061), `hardware.nvidia.modesetting.enable = true;` makes SDDM start on NVIDIA proprietary driver 555.58 + kernel 6.9.7, but the session ends up being unstable for me as explained [here](https://github.com/NixOS/nixpkgs/issues/323396#issuecomment-2198287487).

**Comment by pillowtrucker** [comment]
I have had this issue ever since plasma6 initially released, but I've also had `hardware.nvidia.modesetting.enable = true;` set much longer than that. This doesn't help or work on hybrid setups with AMD iGPU + Nvidia. I am also not using wayland, my system is all xorg. Has anyone looked into the actual sddm module and/or package ? Random workarounds may incidentally work on some computers, but they don't seem to address or even identify the underlying issue here.

**Comment by SuperSandro2000** [comment]
Please see #323396

**Comment by tjkirch** [comment]
> Please see #323396

Isn't that a separate issue?

**Comment by opl-** [comment]
When I created the newer #323396 issue, it was to track SDDM completely crashing, showing the user the journal logs. My understanding at the time was that the symptoms for this issue (#311450) are that the SDDM Greeter starts, but leaves the user on a black screen with nothing but a cursor. I only just noticed the cursor has never been mentioned in this issue.

Still, since both problems manifested differently and were fixed with different config changes for me, it seems to me like these are two different problems with different root causes.
